internal/benchrunner/runners/system/servicedeployer/compose.go

```diff
@@ -0,0 +1,214 @@
+	}
+
+	// Clean service logs
+	err = files.RemoveContent(outCtxt.Logs.Folder.Local)
+	if err != nil {
+		return nil, errors.Wrap(err, "removing service logs failed")
+	}
+
+	serviceName := inCtxt.Name
+	opts := compose.CommandOptions{
+		Env:       []string{fmt.Sprintf("%s=%s", ServiceLogsDirEnv, outCtxt.Logs.Folder.Local)},
+		ExtraArgs: []string{"--build", "-d"},
+	}
+	err = p.Up(opts)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not boot up service using Docker Compose")
+	}
+	err = p.WaitForHealthy(opts)
+	if err != nil {
+		processServiceContainerLogs(p, compose.CommandOptions{
+			Env: opts.Env,
+		}, outCtxt.Name)
+		return nil, errors.Wrap(err, "service is unhealthy")
+	}
+
+	// Build service container name
+	outCtxt.Hostname = p.ContainerName(serviceName)
+
+	// Connect service network with stack network (for the purpose of metrics collection)
+	err = docker.ConnectToNetwork(p.ContainerName(serviceName), stack.Network())
+	if err != nil {
+		return nil, errors.Wrapf(err, "can't attach service container to the stack network")
+	}
+
+	logger.Debugf("adding service container %s internal ports to context", p.ContainerName(serviceName))
+	serviceComposeConfig, err := p.Config(compose.CommandOptions{
+		Env: []string{fmt.Sprintf("%s=%s", ServiceLogsDirEnv, outCtxt.Logs.Folder.Local)},
+	})
+	if err != nil {
+		return nil, errors.Wrap(err, "could not get Docker Compose configuration for service")
+	}
+
+	s := serviceComposeConfig.Services[serviceName]
+	outCtxt.Ports = make([]int, len(s.Ports))
+	for idx, port := range s.Ports {
+		outCtxt.Ports[idx] = port.InternalPort
+	}
+
+	// Shortcut to first port for convenience
+	if len(outCtxt.Ports) > 0 {
+		outCtxt.Port = outCtxt.Ports[0]
+	}
+
+	outCtxt.Agent.Host.NamePrefix = "docker-fleet-agent"
+	service.ctxt = outCtxt
+	return &service, nil
+}
+
+// Signal sends a signal to the service.
+func (s *dockerComposeDeployedService) Signal(signal string) error {
+	p, err := compose.NewProject(s.project, s.ymlPaths...)
+	if err != nil {
+		return errors.Wrap(err, "could not create Docker Compose project for service")
+	}
+
+	opts := compose.CommandOptions{
+		Env:       []string{fmt.Sprintf("%s=%s", ServiceLogsDirEnv, s.ctxt.Logs.Folder.Local)},
+		ExtraArgs: []string{"-s", signal},
+	}
+	if s.ctxt.Name != "" {
+		opts.Services = append(opts.Services, s.ctxt.Name)
+	}
+
+	return errors.Wrapf(p.Kill(opts), "could not send %q signal", signal)
+}
+
+// TearDown tears down the service.
+func (s *dockerComposeDeployedService) TearDown() error {
+	logger.Debugf("tearing down service using Docker Compose runner")
+	defer func() {
+		err := files.RemoveContent(s.ctxt.Logs.Folder.Local)
+		if err != nil {
+			logger.Errorf("could not remove the service logs (path: %s)", s.ctxt.Logs.Folder.Local)
+		}
+	}()
+
+	p, err := compose.NewProject(s.project, s.ymlPaths...)
+	if err != nil {
+		return errors.Wrap(err, "could not create Docker Compose project for service")
```

Comment:
For errors, it could be used `fmt.Errorf` instead of using `errors.Wrap`

```suggestion
		return fmt.Errorf("could not create Docker Compose project for service: %w", err)
```